### PR TITLE
Revise NotEqual operation reference implementation

### PR DESCRIPTION
**Pull request by nsemaev**
### Details:
 - *Restoring the correct specification structure*

### Tickets:
 - *37459*

### CODE REVIEW

File: inference-engine/tests/functional/plugin/shared/include/behavior/preprocessing.hpp [hunk 1]
```diff
@@ -110,9 +110,12 @@ struct PreprocessingPrecisionConvertTest :
     }
 
     void Validate() override {
+        // w/a: copy of original function is required to provide correct op coverage report (overflow of convert counter issue)
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/shared/include/behavior/preprocessing.hpp:113]: Extra changes. Please remove the changes from the pr

Comment (nsemaev) [reply inference-engine/tests/functional/plugin/shared/include/behavior/preprocessing.hpp]: Fixed

File: ngraph/core/reference/include/ngraph/runtime/reference/not_equal.hpp [hunk 2]
```diff
@@ -21,17 +21,18 @@ namespace ngraph
...
-                           char* out,
-                           size_t count) // TODO: using char for bool, is this right?
-            {
-                for (size_t i = 0; i < count; i++)
-                {
-                    out[i] = arg0[i] != arg1[i];
-                }
-            }
+            // not using
+//            template <typename T>
```
Comment (ilyachur) [inline ngraph/core/reference/include/ngraph/runtime/reference/not_equal.hpp:25]: I propose to remove unused methods.

Comment (nsemaev) [reply ngraph/core/reference/include/ngraph/runtime/reference/not_equal.hpp]: Of course. Done

**Pull request closed without merging**